src/fluent/watch.ts

```diff
@@ -382,53 +424,66 @@ export class Watcher<T extends GenericClass> {
 
         const { body } = response;
 
+        if (!body) {
+          throw new Error("No response body found");
+        }
+
         // Reset the retry count
         this.#resyncFailureCount = 0;
         this.#events.emit(WatchEvent.INC_RESYNC_FAILURE_COUNT, this.#resyncFailureCount);
 
+        // Use a native stream issue #1180
+        this.#stream = Readable.from(body);
+        const decoder = new TextDecoder();
+        let buffer = "";
+
         // Listen for events and call the callback function
-        this.#stream.on("data", async line => {
+        this.#stream.on("data", chunk => {
           try {
-            // Parse the event payload
-            const { object: payload, type: phase } = JSON.parse(line) as {
-              type: WatchPhase;
-              object: InstanceType<T>;
-            };
-
-            // Update the last seen time
-            this.#lastSeenTime = Date.now();
-
-            // If the watch is too old, remove the resourceVersion and reload the watch
-            if (phase === WatchPhase.Error && payload.code === 410) {
-              throw {
-                name: "TooOld",
-                message: this.#resourceVersion!,
-              };
+            buffer += decoder.decode(chunk, { stream: true });
+            const lines = buffer.split("\n");
+            buffer = lines.pop()!;
+
+            for (const line of lines) {
+              try {
```

Comment:
nested try/catch can easily turn into bugs, probably best to pull this into it's own function if this is needed

Reply:
sounds good